pkg/controller/cassandradatacenter/statefulset.go

```diff
@@ -24,21 +23,9 @@ const (
 	OperatorConfigVolumeMountPath = "/tmp/operator-config"
 	UserConfigVolumeMountPath     = "/tmp/user-config"
 	UserSecretVolumeMountPath     = "/tmp/user-secret-config"
+	BackupSecretVolumeMountPath   = "/tmp/backup-secret"
```

Comment:
What's the difference between `BackupSecret` and `UserSecret`?

Reply:
The difference is in fact that `BackupSecret` would hold credentials for backup functionality (i.e. GCP) and be mounted in sidecar container, while `UserSecret` is for C* encryption/auth credentials and to be mounted in the Cassandra container. Having backup stuff mounted in the cassandra container or cassandra credentials in the sidecar container, while not horrible, feels pretty hacky and dirty. That's why the split.

Reply:
Maybe change to sidecarSecret/cassandraSecret? So it relates to the pod name

I don't mind the C* and sidecar containers sharing secrets to be honest.

Reply:
Well, I can rename it, but I think it's better called a "backup" secret as it hints the usage for backups only.

Actually, the only other "secrets" usage would be for secure communications as far as I can tell, and we're going to resolve it differently anyway. So I'd just keep it as "backup" for now, and if ever we combine "user" and "backup" secrets, we can change the naming convention.